### DISCUSSION TITLE
Buzzer Functionality in Crazily Firmware

### DISCUSSION THREAD

**Discussion by philwa9** [bitcraze/discussions#1030] [Q&A]
Hello everyone,

I have two specific questions related to the Crazyflie firmware:

1. Does the Crazyflie standard configuration include an onboard buzzer, or is it exclusively available with the use of a Buzzer Deck?

2. I can't locate the initialization process for the buzzer subsystem in the Crazyflie firmware. I located the buzzerInit function declaration in buzzer.h, the corresponding implementation in buzzer.c seems to be empty. Where can I find the specific location where the buzzerInit function is implemented?

I want to use the buzzer functionality to modify it and call it at a specific time in my program loop.

Thanks for your help.

**Comment by knmcguire** [comment]
Hi! Let me try to help out here.

1. If you mean if a crazyflie has a buzzer onboard: no it does not. It does create a startup sound with the motors though but that can't be used during flight. 
2. The [buzzer deck](https://www.bitcraze.io/products/buzzer-deck/) itself has a one wire memory with a specific address that indicates the buzzerdeck. The crazyflie-firmware will dynamically select the [initialization functionality](https://github.com/bitcraze/crazyflie-firmware/blob/dbb9df1137f11d4e7e3771c56d25a7137b5b69cc/src/deck/drivers/src/buzzdeck.c#L55) for that deck. That uses the buzzer.c functions, but the initialization function doesn't have anything to do indeed. The true initialization in done in buzzdeck.c. Tunes for the buzzer can be found in [sound_cf2.c](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/sound_cf2.c)
